### ISSUE TITLE
AudioBufferSourceNode: Clarify offset truncation for negative playbackRates

### ISSUE THREAD

**Issue by collares** [WebAudio/web-audio-api#2031]
**Describe the issue**
The spec says "If offset is greater than loopEnd, playback will begin at loopEnd (and immediately loop to loopStart)". If I understand the intent of the spec correctly, then this applies only when playbackRate is non-negative. The similar instruction "For negative playbackRate, if offset is less than loopStart, playback will begin at loopStart (and immediately loop to loopEnd)" applies in the other case.

**Where Is It**
In the offset argument for https://webaudio.github.io/web-audio-api/#dom-audiobuffersourcenode-start

**Comment by padenot** [comment]
This is #2033 but backward isn't it? It makes sense to fix it in the same way (but in the opposite direction of course).

**Comment by collares** [comment]
This issue is about the following asymmetry: 

- When `playbackRate` is positive, it is possible to start before `loopStart` and play audio until you reach the loop for the first time.
- Therefore, when `playbackRate` is negative, I would expect the behavior to be symmetric: It should be possible to start after `loopEnd` and let the audio play until reaching the loop. But, currently, the spec says that `offset > loopEnd` results in immediate truncation. I believe this makes sense for the positive playbackRate case, but not for the negative one.

In other words, #2033 deals with "positive `playbackRate`, playhead after `loopEnd`". The symmetric version of #2033 would be "negative `playbackRate`, playhead before `loopStart"`. The current issue refers to "negative `playbackRate`, playhead after `loopEnd`". 

Solutions for #2033 that remove the (currently specified) `offset` parameter truncation should fix this issue as well, but #2033 at the moment refers to dynamically-changing loop bounds only, for which the truncation behavior is currently unspecified.

**Comment by padenot** [comment]
I had opened #2072 that was related, and opened two PRs for this.
